WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/SinglePackageHazmatDeclaration.swift

```diff
@@ -0,0 +1,127 @@
+import SwiftUI
+
+struct SinglePackageHazmatDeclaration: View {
+    @ObservedObject private var viewModel: ShippingLabelSinglePackageViewModel
+    @State private var isShowingHazmatSelection = false
+
+    private let safeAreaInsets: EdgeInsets
+
+    init(safeAreaInsets: EdgeInsets,
+         viewModel: ShippingLabelSinglePackageViewModel) {
+        self.safeAreaInsets = safeAreaInsets
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Spacer()
+            VStack {
+                Divider()
+
+                TitleAndToggleRow(title: Localization.containsHazmatMaterials, isOn: $viewModel.containsHazmatMaterials)
+                    .padding(.horizontal, Constants.horizontalPadding)
+
+                VStack {
+                    Divider()
+                        .padding(.horizontal, insets: safeAreaInsets)
+                        .padding(.leading, Constants.horizontalPadding)
+
+                    TitleAndValueRow(title: Localization.hazmatCategoryTitle, value: .placeholder(Localization.selectHazmatCategory)) {
+                        isShowingHazmatSelection.toggle()
+                    }
+
+                    Divider()
+                        .padding(.horizontal, insets: safeAreaInsets)
+                        .padding(.leading, Constants.horizontalPadding)
+
+                    createHazmatInstructionsView()
+
+                }
+                .renderedIf(viewModel.containsHazmatMaterials)
+
+                Divider()
+            }
+            .background(Color(.listForeground(modal: false)))
+
+            Text(Localization.hazmatTooltip)
+                .renderedIf(!viewModel.containsHazmatMaterials)
+                .padding(.leading, Constants.horizontalPadding)
+                .calloutStyle()
+        }
+        .renderedIf(viewModel.isHazmatShippingEnabled)
+    }
+
+    func createHazmatInstructionsView() -> some View {
```

Comment:
super nit: this function can be private like in a private extension

Reply:
Oh, thank you for the suggestion! I thought that View returning functions didn't required visibility changes, I'll make sure to fix that for the other SwiftUI functions I declared. Thanks! Fixed in https://github.com/woocommerce/woocommerce-ios/pull/10552/commits/940b3c8d7c89e8e488a309c679a9bd0d92e75f46